lib/active_merchant/billing/gateways/secure_trading.rb

```diff
@@ -0,0 +1,282 @@
+require 'nokogiri'
+
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    # For more information on the Secure Trading visit
+    # {SecureTraining}[https://docs.securetrading.com]
+    #
+    class SecureTradingGateway < Gateway
+      self.test_url = 'https://webservices.securetrading.net:443/xml/'
+      self.live_url = 'https://webservices.securetrading.net:443/xml/'
+
+      self.supported_countries = ['US']
+      self.default_currency = 'USD'
+      self.supported_cardtypes = [:visa, :master, :american_express, :discover]
+
+      self.homepage_url = 'https://www.securetrading.com/'
+      self.display_name = 'Secure Trading Gateway'
+
+      STANDARD_ERROR_CODE_MAPPING = {
+        '10103' => STANDARD_ERROR_CODE[:incorrect_number],
+        '31001' => STANDARD_ERROR_CODE[:invalid_number],
+        '31002' => STANDARD_ERROR_CODE[:expired_card],
+        '31003' => STANDARD_ERROR_CODE[:expired_card],
+        '31004' => STANDARD_ERROR_CODE[:expired_card],
+        '31009' => STANDARD_ERROR_CODE[:processing_error],
+        '70000' => STANDARD_ERROR_CODE[:card_declined],
+      }
+
+      def initialize(options={})
+        requires!(options, :api_key, :user_id, :site_id)
+        @api_key = options[:api_key]
+        @user_id = options[:user_id]
+        @site_reference = options[:site_id]
+        super
+      end
+
+      def authorize(money, payment, options={})
+        parameter = Nokogiri::XML::Builder.new(encoding: 'UTF-8') do |xml|
+          xml.requestblock version: '3.67' do
```

Comment:
Maybe you could DRY this a bit and extract the `Nokogiri::XML::Builder` to a method and re-use that in this `authorize` and the `purchase` methods?